lib/assets/javascripts/utils/ariatiseForm.js

```diff
@@ -44,48 +44,52 @@
     3. Prevent form to be submitted
 */
 (function(ctx){
-    var requiredFields=(function(selector){
-        return $(selector).find('.form-control').filter('[aria-required="true"]');
+    // Collect all elements that are either required or have data-validation defined
+    var validatableFields=(function(selector){
+        return $(selector).find('.form-control').filter('[data-validation],[aria-required="true"]');
     });
     var blockHelp=(function (id,type){
-        var msg='Please fill out this field with a valid '+type+'.';    //TODO internationalisation
-        return '<span id="'+id+'" class="help-block" style="display:none;">'+msg+'</span>';
+      return '<span id="'+id+'" class="help-block" style="display:none;">'+defaultValidationError(type)+'</span>';
     });
     var ariaDescribedBy=(function(value){
         return { 'aria-describedby': value };
     });
     var ariaInvalid=(function(value){
         return { 'aria-invalid': value };
     });
+    var defaultValidationError=(function(type){
+      if(dmproadmap.utils.validate[type] && dmproadmap.utils.validate[type].message){
+        return dmproadmap.utils.validate[type].message;
+      }
+      return '';
+    });
     var validationStates={
             hasWarning: 'has-warning',
             hasError: 'has-error',
             hasSuccess: 'has-success'
     };
-    var getTypeForSubmittableElement=(function(el){
-        // Reference from https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories#Form-associated_content
-        if($(el).is('input')){
-            return $(el).attr('type');  // available types at https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Form_<input>_types
-        }
-        else if($(el).is('select')){
-            return 'select';
-        }
-        else if($(el).is('textarea')){
-            return 'textarea';
-        }
-        else
-            return 'unknown';
+    var getValidationTypeForElement=(function(el){
+      var validation = $(el).attr('data-validation');
+      // if the specified validation type is defined
+      if(validation && dmproadmap.utils.validate[validation]){
+        return $(el).attr('data-validation');
+
+      }else if($(el).attr('aria-required') === 'true'){
+        return 'required';
```

Comment:
Are you assuming that absence of data-validation attribute should fallback to type 'required'. This is too strong assumption that I can think cases that would fail such as expecting a number, or an email... Absence of data-validation, should fallback to the default validation defined for the type, like it was before adding data-validation.